Google-Summer-of-Code-2023.md

```diff
@@ -273,3 +273,504 @@ The following is a list of Oppia's 2023 GSoC project ideas. You are welcome to c
+
+Additionally, finish adding [computer-aided translation (CAT)](https://docs.google.com/document/d/1kJd-yLTzB9a2c3Nq7v9pzKfHwKHKGpkWfQ8B0YGf50U/edit#heading=h.24wysknhgyrz) support when submitting translation suggestions. 
+
+**Size of this project:** medium (~175 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:**
+
+* Knowledge and understanding of Python
+* Knowledge and understanding of TypeScript, Angular, basic HTML, and CSS
+
+**Suggested Milestones:**
+
+* **Milestone 1:** Complete the backend framework for prioritized languages:
+  - Add notion of prioritized language in backend storage schemas
+  - Store list of prioritized languages
+  - Create beam job to auto-translate remaining text content for a prioritized language
+
+
+* **Milestone 2:** Surface prioritized languages in the UI and add CAT support:
+  - Enable the adding of a prioritized language in the contributor dashboard admin page
+  - Add note to lesson text content that has been auto-translated
+  - Auto-translate non-prioritized language once translation threshold has been achieved (95%)
+  - Add computer-aided translation capability to translation contribution modal
+
+
+
+**Dependency on Release Schedule:** 
+
+**Proposal notes:**
+
+* We’ll likely write a beam job to populate the auto-translations for a language. See the [wiki](https://github.com/oppia/oppia/wiki/Apache-Beam-Jobs) for more details. 
+* Maybe ~80% of the CAT work is already done, including all of the backend changes. We just need to hook it up in the frontend, e.g. see working PR: [#12624](https://github.com/oppia/oppia/pull/12604/files) and related issue [#11434](https://github.com/oppia/oppia/issues/11434). 
+  - See Computer Aided Translations (CAT) backend milestone PRs.
+* The CAT backend utilizes the Google Translate API for auto-translations. We can re-use that work here for auto-translations. Alternatively, we can investigate other translate APIs that may offer better translations, e.g. Oppia has done studies in the past that have shown Microsoft to produce preferred translations.
+
+
+**Useful resources:**
+
+### 2.3. Improvements for translation reviewer experience
+
+**Project Description:**
+
+Translation reviewers have given feedback such that the current contributor dashboard reviewer workflow doesn’t match with their most of the needs. This project aims to improve the translation reviewer experience in the contributor dashboard.
+**Size of this project:** large (~350 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:**
+
+* Knowledge and understanding of Python
+* Knowledge and understanding of TypeScript, Angular, HTML, and CSS
+
+**Suggested Milestones:**
+
+* **Milestone 1:**
+  - Showing translation cards according to the order of the content flow in the lesson. Exploration editor should also be linked so that the original lesson can be seen.
```

Comment:
```suggestion
  - Show translation cards in the same order as content flows in the lesson. The appropriate exploration editor should also be linked so that the original lesson can be seen.
```

What if the translator doesn't have permission to edit the lesson? They won't be able to see the editor, will they?